toolchain.mk

```diff
@@ -23,10 +23,13 @@ SRC_AARCH64_GCC 		?= https://developer.arm.com/-/media/Files/downloads/gnu-a/10.
 # $(3) is $.._GCC_VERSION	: the name of the file to download
 define dltc
 	@if [ ! -d "$(1)" ]; then \
-		mkdir -p $(1); \
 		echo "Downloading $(3) ..."; \
-		curl -s -L $(2) -o $(TOOLCHAIN_ROOT)/$(3).tar.xz; \
-		tar xf $(TOOLCHAIN_ROOT)/$(3).tar.xz -C $(1) --strip-components=1; \
+		curl --retry-all-errors --retry 5 -s -L $(2) -o $(TOOLCHAIN_ROOT)/$(3).tar.xz || \
+			{ rm -f $(TOOLCHAIN_ROOT)/$(3).tar.xz; echo Download failed; exit 1; }; \
+		mkdir -p $(1); \
```

Comment:
This must be moved back to where is was before (before `curl`).

Reply:
You mean mkdir? It creates directory like aarch32, that doesn't needed for download process.
If move it back, we need removing this directory in case of download errors.

Reply:
Yes. `mkdir -p $(1)` also creates the parent directory where `curl` is supposed to download the archive. You have tested this patch, right? ;-)

Reply:
Curl download file to the parent directory. This patch was tested :)

Reply:
Doesn't work for me...
```
jerome@matebook:~/work/optee_repo_hikey/build (master)$ git fetch github pull/508/head
remote: Enumerating objects: 15, done.
remote: Counting objects: 100% (15/15), done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 15 (delta 8), reused 13 (delta 6), pack-reused 0
Unpacking objects: 100% (15/15), 5.13 KiB | 404.00 KiB/s, done.
From https://github.com/OP-TEE/build
 * branch            refs/pull/508/head -> FETCH_HEAD
jerome@matebook:~/work/optee_repo_hikey/build (master)$ git co FETCH_HEAD 
Note: switching to 'FETCH_HEAD'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 7259e9e Update toolchain.mk
jerome@matebook:~/work/optee_repo_hikey/build ((7259e9e...))$ make toolchains
Downloading gcc-arm-10.2-2020.11-x86_64-arm-none-linux-gnueabihf ...
curl: (23) Failure writing output to destination
Download failed
make: *** [toolchain.mk:42: aarch32] Error 1
jerome@matebook:~/work/optee_repo_hikey/build ((7259e9e...))$
```

Reply:
Hm... I understand: toolchain directory doesn't exists before first build and created with subdirectories. Will fix it.